railgun/test/integration/suite_test.go

```diff
@@ -73,9 +80,63 @@ func deployControllers(ctx context.Context, ready chan ktfkind.ProxyReadinessEve
 		u := event.URL
 		proxyReady <- u
 
-		cmd := exec.CommandContext(ctx, "go", "run", "../../main.go", "--kong-url", fmt.Sprintf("http://%s:8001", u.Hostname()))
+		// create a tempfile to hold the cluster kubeconfig that will be used for the controller
+		kubeconfig, err := ioutil.TempFile(os.TempDir(), "kubeconfig-")
+		if err != nil {
+			panic(err)
+		}
+		defer kubeconfig.Close()
+
+		// dump the kubeconfig from kind into the tempfile
+		generateKubeconfig := exec.CommandContext(ctx, "kind", "get", "kubeconfig", "--name", cluster.Name())
+		generateKubeconfig.Stdout = kubeconfig
+		generateKubeconfig.Stderr = os.Stderr
+		if err := generateKubeconfig.Run(); err != nil {
+			panic(err)
+		}
+
+		// set the default command which runs the current controller manager code
+		cmd := exec.CommandContext(ctx, "go", "run", "../../main.go",
+			"--kong-url", fmt.Sprintf("http://%s:8001", u.Hostname()),
+			"--kubeconfig", kubeconfig.Name())
+
+		// if set, allow running the legacy controller for the tests instead
+		// TODO: this will be removed as part of KIC 2.0, where the legacy controller will be replaced.
+		//       for more details see the relevant milestone: https://github.com/Kong/kubernetes-ingress-controller/milestone/12
+		if os.Getenv(LegacyControllerEnvVar) != "" {
```

Comment:
Please wrap this in an abstraction to minimize side effects and local variables:

```go
useLegacyKIC := (os.Getenv(...) != "")

var kicCmd *exec.Cmd
if legacyKIC {
  cmd = buildKICCommandLegacy(cluster.Client())
} else {
  cmd = buildKICCommandRailgun()
}
```

Reply:
f2a6e4195943b34ae700fbc5c661b49a97e62efc is intended to resolve this :+1:

Reply:
Hmm... OK. I don't feel strongly about this either way, so I'll simply change it to the way you like :+1:

Reply:
39a8d4498ea2d2a736ef3d474ad33008ef316542 should resolve this, let me know if that looks good to you!